graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_numerical_transformation.py

```diff
@@ -0,0 +1,349 @@
+
+        vector_cols = [col + "_vec" for col in cols]
+        scaled_cols = [col + "_scaled" for col in cols]
+        pipeline = Pipeline(stages=assemblers + scalers)
+        scaler_model = pipeline.fit(imputed_df)
+        scaled_df = scaler_model.transform(imputed_df).drop(*vector_cols).drop(*cols)
+
+        scaled_df = scaled_df.select(
+            *[
+                (vec_udf(F.col(scaled_col_name))).alias(orig_col)
+                for scaled_col_name, orig_col in zip(scaled_cols, cols)
+            ]
+            + other_cols
+        )
+    elif shared_norm == "standard":
+        col_sums = imputed_df.agg({col: "sum" for col in cols}).collect()[0].asDict()
+        # TODO: See if it's possible to exclude NaN values from the sum
+        for col, val in col_sums.items():
+            if np.isinf(val) or np.isnan(val):
+                col_sums[col] = 0
+        scaled_df = imputed_df.select(
+            [(F.col(c) / col_sums[f"sum({c})"]).alias(c) for c in cols] + other_cols
+        )
+    else:
+        raise RuntimeError(f"Uknown normalizer type requested for cols {cols}: {shared_norm}")
+
+    return scaled_df
+
+
+class DistNumericalTransformation(DistributedTransformation):
+    """Transformation to apply missing value imputation and various forms of normalization
+     to a numerical input.
+
+    Parameters
+    ----------
+    cols : Sequence[str]
+        The list of columns to apply the transformations on.
+    normalizer : str
+        The normalization to apply to the columns.
+        Valid values are "none", "min-max", and "standard".
+    imputer : str
+        The type of missing value imputation to apply to the column.
+        Valid values are "mean", "median" and "most_frequent".
+    """
+
+    def __init__(self, cols: Sequence[str], normalizer: str, imputer: str) -> None:
+        super().__init__(cols)
+        self.cols = cols
+        self.shared_norm = normalizer
+        # Spark uses 'mode' for the most frequent element
+        self.shared_imputation = "mode" if imputer == "most_frequent" else imputer
+
+    def apply(self, input_df: DataFrame) -> DataFrame:
+        logging.debug(
+            "Applying normalizer: %s, imputation: %s", self.shared_norm, self.shared_imputation
+        )
+
+        imputed_df = apply_imputation(self.cols, self.shared_imputation, input_df)
+        scaled_df = apply_norm(self.cols, self.shared_norm, imputed_df)
+
+        # TODO: Figure out why the transformation is producing Double values, and switch to float
+        return scaled_df
+
+    @staticmethod
+    def get_transformation_name() -> str:
+        return "DistNumericalTransformation"
+
+
+class DistMultiNumericalTransformation(DistNumericalTransformation):
+    """Transformation to apply missing value imputation and various forms of normalization
+     to a multi-column numerical input, where the input is a string separated by a delimiter.
```

Comment:
Are we going to have one imputer for each column and one normalizer for each column?

Reply:
Ah I see this reads as if we do. I'll change the language.

To answer the question in any case, was not planning on supporting separate imputers/normalizers for each col to at least for this iteration. We could implement it if we have a user request.